src/datadog/telemetry/telemetry.cpp

```diff
@@ -1,176 +1,119 @@
-#include <datadog/clock.h>
-#include <datadog/datadog_agent_config.h>
-#include <datadog/dict_writer.h>
-#include <datadog/runtime_id.h>
-#include <datadog/string_view.h>
+#include <datadog/optional.h>
 #include <datadog/telemetry/telemetry.h>
 
-#include <chrono>
+#include "telemetry_impl.h"
 
-#include "datadog_agent.h"
-#include "platform_util.h"
-#include "tracer_telemetry.h"
+namespace datadog::telemetry {
+namespace details {
```

Comment:
nit: why did you choose details ? May I suggest changing for helpers ?

Reply:
the `details` namespace is a comment convention in cpp for internal details

Reply:
I did not know, thank you for the discovery